rate_based_rule/main.tf

```diff
@@ -1,19 +1,24 @@
 resource "aws_wafregional_byte_match_set" "url_match" {
-  name = "wafregional_byte_match_set"
+  name  = "wafregional_byte_match_set"
+  count = length(var.allowed_headers) >= 1 ? 1 : 0
 
-  byte_match_tuples {
-    text_transformation   = "LOWERCASE"
-    target_string         = var.pattern
-    positional_constraint = var.positional_constraint
-
-    field_to_match {
-      type = "URI"
+  dynamic "byte_match_tuples" {
+    for_each = var.allowed_headers
+    content {
+      field_to_match {
+        type = lookup(byte_match_tuples.value, "type", null)
+        data = lookup(byte_match_tuples.value, "data", null)
+      }
+      target_string         = lookup(byte_match_tuples.value, "target_string", null)
+      positional_constraint = lookup(byte_match_tuples.value, "positional_constraint", null)
+      text_transformation   = lookup(byte_match_tuples.value, "text_transformation", null)
     }
   }
 }
 
 resource "aws_wafregional_rate_based_rule" "url_match_rule" {
-  name        = "wafregional_rate_based_match"
+  name  = "wafregional_rate_based_match"
+  count = length(var.allowed_headers) >= 1 ? 1 : 0
   tags        = local.tags
   metric_name = var.metric_name
```

Comment:
same here

Reply:
updated the changes.
Thanks for noticing